### PR TITLE
chore: Enable DCM rule avoid-unnecessary-type-assertions

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

Enable DCM rule avoid-unnecessary-type-assertions.

More details [here](https://dcm.dev/docs/rules/common/avoid-unnecessary-type-assertions/). Note that we have to ignore mixins otherwise the rule has false positives.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org/
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame_lint/lib/analysis_options_with_dcm.yaml [hunk 1]
```diff
@@ -11,5 +11,7 @@ dart_code_metrics:
     - avoid-top-level-members-in-tests:
         include:
           - 'test/**/*_test.dart'
+    - avoid-unnecessary-type-assertions:
+      ignore-mixins: true
```
Comment (erickzanardo) [inline packages/flame_lint/lib/analysis_options_with_dcm.yaml:15]: Should this line have a dash too?

Comment (luanpotter) [reply packages/flame_lint/lib/analysis_options_with_dcm.yaml:15]: No that is a parameter

Comment (erickzanardo) [reply packages/flame_lint/lib/analysis_options_with_dcm.yaml:15]: Oh that makes sense, I totally got confused

**Review by erickzanardo — APPROVED** [review]

File: packages/flame_lint/lib/analysis_options_with_dcm.yaml (hunk 1 above)
Comment (luanpotter) [reply packages/flame_lint/lib/analysis_options_with_dcm.yaml:15]: yeah yaml can get weird for lists of keys mixed with objects

**Pull request merged**